crates/bevy_ecs/src/storage/sorted_small_vec.rs

```diff
@@ -0,0 +1,259 @@
+use smallvec::SmallVec;
+use core::cmp::Ordering;
+
+/// Stores a sorted list of indices with quick implementation for union, difference, intersection.
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct SortedSmallVec<const N: usize>(SmallVec<[usize; N]>);
```

Comment:
Why is this `pub`? Why not `pub(crate)`?

This extends to `SortedSmallVec::new` and `SortedSmallVec::from_vec`